### PR TITLE
chain: Obi go lib test coverage 💯

### LINKED ISSUE #1718: Add unit test for OBI go lib

### PR DESCRIPTION
**Pull request by perimeko**
Fixed #1718

### CODE REVIEW

File: chain/pkg/obi/schema_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+type EmptySchema struct{}
+
+type Color struct {
+	code string `code` //missing obi
```
Comment (sorawit) [inline chain/pkg/obi/schema_test.go]: Why is this not `obi:"code"`? Also `Code` should be public.

File: chain/pkg/obi/schema_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
```
Comment (sorawit) [inline chain/pkg/obi/schema_test.go:13]: Can we also add test for `bytes`?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/pkg/obi/schema_test.go (hunk 2 above)
Comment (perimeko) [reply chain/pkg/obi/schema_test.go:13]: Added test case for bytes

Comment (perimeko) [reply chain/pkg/obi/schema_test.go:13]: This struct is expected to be mock missing OBI tag struct.

File: chain/pkg/obi/schema_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+type EmptySchema struct{}
+
+type NoOBITagStruct struct {
+	NoOBITag string `"noOBItag"` //missing obi
+}
+
+type Validation struct {
```
Comment (sorawit) [inline chain/pkg/obi/schema_test.go]: let's call this NotSupportedStruct

**Review by sorawit — CHANGES_REQUESTED** [review]

**Review by sorawit — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**